src/App.tsx

```diff
@@ -15,49 +17,121 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((previousGood, currentGood) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return previousGood.localeCompare(currentGood);
+
+      case SortType.LENGTH:
+        return previousGood.length - currentGood.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed
+    ? visibleGoods.reverse()
+    : visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReverse] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const handleReverseClick = () => {
+    setReverse(current => !current);
+  };
+
+  const sortByType = (type: SortType) => {
+    setSortType(type);
+  };
+
+  const handleResetClick = () => {
```

Comment:
Use one naming strategy: either you start with `handle`, or you start with an action verb.